runtime/tests/checker/entitlements_test.go

```diff
@@ -5132,3 +5132,32 @@ func TestCheckBuiltinEntitlements(t *testing.T) {
 
 	assert.NoError(t, err)
 }
+
+func TestCheckIdentityMapping(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+        struct S {
+            access(Identity) fun foo(): auth(Identity) &AnyStruct {
+                let a: AnyStruct = "hello"
+                return &a as auth(Identity) &AnyStruct
+            }
+        }
+
+        fun main() {
+            let s = S()
```

Comment:
We probably want some more tests here. In particular:

* Test that an unauthorized `&S` maps to an unauthorized `&AnyStruct`
* The identity map preserves conjunctions/disjunctions (i.e. `A, B` yields `A, B` while `C | D` yields `C | D`
* Test the behavior of an identity-mapped member when accessed on an owned value.

Reply:
added 7a4ff32be9dc566cba30726f1ffddf2588ccc3e8